torch/testing/_internal/common_device_type.py

```diff
@@ -843,7 +843,20 @@ def _parametrize_test(self, test, generic_cls, device_cls):
                 try:
                     @wraps(test)
                     def test_wrapper(*args, **kwargs):
-                        return test(*args, **kwargs)
+                        try:
+                            return test(*args, **kwargs)
+                        except Exception as e:
+                            tracked_input = get_tracked_input()
+                            if tracked_input is not None:
+                                raise Exception(
+                                    f"Error caused by {tracked_input.input_type} "
+                                    f"{tracked_input.index}: {tracked_input.input_val}") from e
+                            raise e
+
+                    # Initialize info for the last inputs seen. This is useful for tracking
+                    # down which inputs caused a test failure.
+                    # Mapping format: test ID -> TrackedInput
```

Comment:
Why is this needed? Aren't you only ever running one test at a time anyways?

Reply:
This was written under the assumption that we could be running tests across multiple threads, but I'm not sure if that's true?

Reply:
yes thats correct we do run tests in parallel across different processes